tests/test_render.py

```diff
@@ -2257,3 +2258,324 @@ def test_clientside_fails_when_returning_a_promise(self):
         self.wait_for_text_to_equal('#input', 'hello')
         self.wait_for_text_to_equal('#side-effect', 'side effect')
         self.wait_for_text_to_equal('#output', 'output')
+    """
+
+    def test_devtools_python_errors(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            html.Button(id='python', children='Python exception', n_clicks=0),
+            html.Div(id='output')
+        ])
+
+        @app.callback(
+            Output('output', 'children'),
+            [Input('python', 'n_clicks')])
+        def update_output(n_clicks):
+            if n_clicks == 1:
+                1/0
+            elif n_clicks == 2:
+                raise Exception('Special 2 clicks exception')
+
+        self.startServer(
+            app,
+            debug=True,
+            use_reloader=False,
+            use_debugger=True,
+            dev_tools_hot_reload=False,
+        )
+
+        self.percy_snapshot('devtools - python exception - start')
+
+        self.wait_for_element_by_css_selector('#python').click()
+        self.percy_snapshot('devtools - python exception - closed')
+        self.wait_for_element_by_css_selector('.test-devtools-error-toggle').click()
+        self.percy_snapshot('devtools - python exception - open')
+
+        self.wait_for_element_by_css_selector('#python').click()
+        self.percy_snapshot('devtools - python exception - 2 errors')
+        self.wait_for_element_by_css_selector('.test-devtools-error-toggle').click()
+        self.percy_snapshot('devtools - python exception - 2 errors open')
```

Comment:
@chriddyp I'm afraid some of the percy snapshots will need continuously approval like: time stamp, library change the line number etc.

![image](https://user-images.githubusercontent.com/1394467/56101688-894bfb00-5ef4-11e9-815d-777bbb4dd925.png)

Reply:
ah yeah :( I For the timestamp, I think I'll give https://docs.percy.io/docs/percy-specific-css a try. For the line numbers, I think they'll change less frequently since the exception remains the same. We could even pin `flask` too.

Reply:
Looking back at this - for the line numbers, it looks like the `test_render` traceback doesn't appear in this screenshot test (its hidden behind the scroll). So, for now I don't think it'll be an issue but it could be in the future if those exceptions become visible.

Reply:
Timestamp fixed in https://github.com/plotly/dash-renderer/pull/156/commits/ec81d67ebdc33ce1e216795962c25efeb0939665